### ISSUE TITLE
 Bug: Section headers at emoji picker menu disappear on scrolling at Firefox reported by @thesahindia 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11851]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open ND in Firefox browser on mobile
2. Navigate to a chat
3. Open emoji picker and scroll

## Expected Result:
The headers shouldn't disappear

## Actual Result:
The section headers disappear.

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Mobile Web - Firefox in Android

**Version Number:** 1.2.15-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @thesahindia 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665695922095369

https://user-images.githubusercontent.com/43996225/195894930-8080a0c9-4b40-43ea-944e-259f71d59ee6.mp4

<img src="https://user-images.githubusercontent.com/43996225/195894963-d27fa440-9823-446b-83b3-5244ec095553.jpg" width="40%" height="40%">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sonialiap** [comment]
I can reproduce following the steps in the OP 👍

**Comment by neil-marcellini** [comment]
Looks like a legit problem that an external contributor could fix.

**Comment by roryabraham** [comment]
@miljakljajic Let's get an Upwork job posted for this issue

**Comment by miljakljajic** [comment]
[Posted](https://www.upwork.com/ab/applicants/1584894080490983424/job-details) to Upwork!

**Comment by miljakljajic** [comment]
Price doubled to 500.

**Comment by miljakljajic** [comment]
Doubled to 1000.

**Comment by wildan-m** [comment]
# Proposal

### Root Cause

By default, [`react-native-web` add `translateZ:0` to `ScrollView` element](https://github.com/necolas/react-native-web/blob/80ef32f6eea4a7950c07b1ea71b45b5c2b0269ee/packages/react-native-web/src/exports/ScrollView/index.js#L311-L314)

This line of code is used to optimize the scroll performance by [tricking the browser to use GPU for transition and animation](https://stackoverflow.com/questions/10814178/css-performance-relative-to-translatez0)

Actually, I’m not sure which one is the root cause between these two:
1. Bug in firefox mobile render engine
2. Bug when trying to combine hardware compositing optimization with the sticky element

### 1. Bug in firefox mobile render engine
Since this issue only occurred on the mobile version of Firefox, there might be a chance that the issue come from Firefox. There are similar issues in their repo. All of the issues are closed, but our case might be an edge case and not resolved yet.

https://github.com/mozilla-mobile/fenix/issues/16772
https://github.com/mozilla-mobile/fenix/issues/19360
https://github.com/mozilla-mobile/fenix/issues/20744

### 2. Bug when trying to combine hardware compositing optimization with the sticky element
Some people reported that adding translate-z-0 will [mess up the sticky elements](https://stackoverflow.com/a/23567225/1174657) and also [cause a memory overload and crash the application](https://stackoverflow.com/a/18728536/1174657). I've tried to untick the `position:sticky` from the emoji header and it shows the header, but we'll lose the sticky header behavior.
![Recording #46](https://user-images.githubusercontent.com/11847279/202220327-42447b32-dc92-461a-a4bb-b7a73451ba8b.gif)

### Workaround

We can exclude `translate-z-0` optimization for Firefox mobile by overriding the `transform` property

```diff
diff --git a/src/components/EmojiPicker/EmojiPickerMenu/index.js b/src/components/EmojiPicker/EmojiPickerMenu/index.js
index 0a2c2c8b1e..33ec04cf9c 100755
--- a/src/components/EmojiPicker/EmojiPickerMenu/index.js
+++ b/src/components/EmojiPicker/EmojiPickerMenu/index.js
@@ -398,6 +398,12 @@ class EmojiPickerMenu extends Component {
         return this.props.isSmallScreenWidth && this.props.windowWidth >= this.props.windowHeight;
     }
 
+    isFirefoxMobile() {
+        const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+        const isMobile = navigator.userAgent.toLowerCase().indexOf('mobile') > -1;
+        return isFirefox && isMobile;
+    }
+
     /**
      * @param {Number} skinTone
      */
@@ -501,6 +507,7 @@ class EmojiPickerMenu extends Component {
                             style={[
                                 styles.emojiPickerList,
                                 this.isMobileLandscape() && styles.emojiPickerListLandscape,
+                                this.isFirefoxMobile() && styles.emojiPickerListFirefoxMobile,
                             ]}
                             extraData={
                               [this.state.filteredEmojis, this.state.highlightedIndex, this.props.preferredSkinTone]

diff --git a/src/styles/styles.js b/src/styles/styles.js
index 2cb23bfa7a..da2c946085 100644
--- a/src/styles/styles.js
+++ b/src/styles/styles.js
@@ -1442,6 +1442,11 @@ const styles = {
         width: '100%',
         ...spacing.ph4,
     },
+    
+    emojiPickerListFirefoxMobile: {
+        transform: [{translateZ: 'unset'}],
+    },
+    
     emojiPickerListLandscape: {
         height: 240,
     },
```

### Result
![Recording #49](https://user-images.githubusercontent.com/11847279/202223848-6b629454-59c3-4cd1-810d-5b6ce504a6bf.gif)

**Comment by miljakljajic** [comment]
@eVoloshchak what do you think of the proposal above?

**Comment by eVoloshchak** [comment]
@wildan-m, cool, thank you for the thorough explanation!
This resolves the issue for me too, latest firefox 07.1.0

However, I'm not sure what we do with such workarounds, this is not fixing the root cause. But I doubt this will be fixed in firefox any time soon, so I think it's worth reporting this bug to firefox and implementing a workaround in the meantime

@roryabraham, what are your thoughts?

**Comment by iwiznia** [comment]
I don't think we want the workaround, especially since:
- This only happens on mobile firefox 
- We are not focusing on supporting firefox right now
- The app is still usable
- The user can also use the phones emoji keyboard

I think we should close.

**Comment by marcaaron** [comment]
Could also see a case for leaving it open but not taking any action for now. We can file a bug report with `react-native-web` at least so it gets on their radar and may get fixed.

**Comment by wildan-m** [comment]
@marcaaron I don't think it's `react-native-web` bug, they added `translate-z-0` on purpose, but the mobile version of FF might have a bug rendering it.

**Comment by roryabraham** [comment]
I personally think we should do what @wildan-m suggested and:

- Report the issue
- Implement the workaround

My reasoning is that if it's a render bug in mobile Firefox, we can't practically solve the issue with a "first class" "non-workaround" solution. So the choices are:

- Don't fix the issue and agree that we do not Firefox
- Fix it with a workaround.

So I'd rather just fix it and make the product better. Don't feel that strongly though because not many people use Firefox.

**Comment by marcaaron** [comment]
I don't think we need to wait on that. Let's just close it and if someone feels like doing that they can spend time on it. 

Remember this is "Firefox on Android" which probably has less than half a percent of the global mobile market share.
